lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,925 @@
+local checkluatype = SF.CheckLuaType
+local checkpattern = SF.CheckPattern
+local registerprivilege = SF.Permissions.registerPrivilege
+
+-- checksafety and assertsafety are here to ensure that even if DarkRP's API changes, it won't create a security vulnerability.
+local whitelist = {
+	["nil"] = true,
+	["boolean"] = true,
+	["number"] = true,
+	["string"] = true,
+}
+local pairs = pairs
+local select = select
+local type = type
+local unpack = unpack
+local function checksafety(...)
+	for k=1, select("#", ...) do
+		local v = select(k, ...)
+		if not whitelist[type(v)] then
+			return false
+		end
+	end
+	return true
+end
+local function assertsafety(...)
+	local args = {...}
+	for k, v in pairs(args) do
+		if not whitelist[type(v)] then
+			args[k] = nil
```

Comment:
Putting nil values into the array will cause the 'unpack' to fail. You need table.remove and a while loop instead.